src/logic.js

```diff
@@ -265,6 +264,21 @@ define([
             });
             return messages;
         },
+        addReferences: function (mug, property, value) {
+            // get absolute paths from mug property's value
+            var isLabel = false,
```

Comment:
That's a sweet refactor https://github.com/dimagi/Vellum/pull/652/commits/455205c500a181168ccf2c672499a82dde0c72de